doc/2/api/controllers/document/upsert/index.md

```diff
@@ -0,0 +1,107 @@
+---
+code: true
+type: page
+title: upsert
+---
+
+# upsert
+
+Applies partial changes to a document. If the document doesn't already exist, a new document is created.
+
+---
+
+## Query Syntax
+
+### HTTP
+
+```http
+URL: http://kuzzle:7512/<index>/<collection>/<_id>/_upsert[?refresh=wait_for][&retryOnConflict=<int>][&source]
+Method: PUT
+Body:
+```
+
+```js
+{
+  changes: {
+    // document partial changes
+  },
+  default: {
+    // optional: document fields to add to the "update" part if the document
+    // is created
+  }
+}
+```
+
+### Other protocols
+
+```js
+{
+  "index": "<index>",
+  "collection": "<collection>",
+  "controller": "document",
+  "action": "update",
+  "_id": "<documentId>",
+  "body": {
+    "changes": {
+      // document partial changes
+    },
+    "default": {
+      // optional: document fields to add to the changes if the document
+      // is created
+    }
+  }
+}
+```
+
+---
+
+## Arguments
+
+- `_id`: unique identifier of the document to update
+- `collection`: collection name
+- `index`: index name
+
+### Optional
+
+- `refresh`: if set to `wait_for`, Kuzzle will not respond until the document is indexed
+- `retryOnConflict`: conflicts may occur if the same document gets updated multiple times within a short timespan, in a database cluster. You can set the `retryOnConflict` optional argument (with a retry count), to tell Kuzzle to retry the failing updates the specified amount of times before rejecting the request with an error.
+- `source`: if set to `true` Kuzzle will return the entire updated document body in the response.
+
+---
+
+## Body properties
+
+- `changes`: partial changes to apply to the document
+- `default`: (optional) fields to add to the document if it gets created
+
+---
+
+## Response
+
+Returns information about the updated document:
+
+- `_created`: if `true`, a new document was created, otherwise the document existed and was updated
```

Comment:
We already have a `created` boolean for the `createOrReplace` action family. IMHO we should keep the same name to ensure consistency in the actual API. (even if I prefer `_created`)

Reply:
Dang. I didn't think about checking the response returned by createOrReplace, it's only through luck that we return the same `created` value (for once this is not intentional :sweat_smile: )

So, for the sake of homogeneity, we indeed need to rename it... It's a shame, I also prefer `_created` over `created.

Reply:
Response property renamed.